lib/LockHandler.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace TwentySixB\Translations;
+
+use ArrayObject;
+
+/**
+ * Handles the lock file, reading and writing.
+ *
+ * @since      0.0.0
+ * @package    TODO:
+ * @subpackage TODO:
+ * @author     TODO:
+ */
+class LockHandler extends ArrayObject {
+
+	/**
+	 * Singleton instance.
+	 *
+	 * @since 0.0.0
+	 * @var   null|LockHandler
+	 */
+	private static $instance = null;
+
+	/**
+	 * Lock file name.
+	 *
+	 * @since 0.0.0
+	 * @var   string
+	 */
+	const FILENAME = 'midoru.lock';
```

Comment:
Otherwise, it's only "middleman" 😄 
```suggestion
	const FILENAME = 'i18n-midoru.lock';
```